src/order/disjointed.lean

```diff
@@ -7,84 +7,90 @@ import order.partial_sups
 
 /-!
 # Consecutive differences of sets
-
 This file defines a way to make a sequence of sets into a sequence of disjoint sets.
-
 ## Main declarations
-
 * `disjointed f`: Yields the sequence of sets `f 0`, `f 1 \ f 0`, `f 2 \ (f 0 ∪ f 1)`, ... This
   sequence has the same union as `f 0`, `f 1`, `f 2` but with disjoint sets.
 -/
 
 variables {α β : Type*}
 
-namespace set
+section generalized_boolean_algebra
+variables [generalized_boolean_algebra α]
 
-/-- If `f : ℕ → set α` is a sequence of sets, then `disjointed f` is
+/-- If `f : ℕ → α` is a sequence of sets, then `disjointed f` is
   the sequence formed with each set subtracted from the later ones
   in the sequence, to form a disjoint sequence. -/
-def disjointed (f : ℕ → set α) (n : ℕ) : set α := f n ∩ (⋂ i < n, (f i)ᶜ)
-
-variables {f : ℕ → set α} {n : ℕ}
+def disjointed (f : ℕ → α) : ℕ → α
```

Comment:
What's the motive for this inductive definition and not the supr one?

Reply:
We don't have a `supr` in a generalized boolean algebra.

Reply:
I don't see a proof that this definition is equivalent to the old one (on complete boolean algebras). Am I missing it, or is it not there? If the latter, could you add one?

Reply:
Sure. It's also easy to get from rewriting `partial_sups`. The exact form is actually never used. The only use of `disjointed` in mathlib currently is to bound the measure of `⋃ n, f n` by the sum of the measures of the `f n`. For that, you just need `Union_disjointed_eq` and `disjointed_le`.